script.js

```diff
@@ -1,12 +1,208 @@
+  allShows.forEach((show) => {
+    output += `
+      <div class="col" style="height: 550px">
+        <div class="card h-100 px-0 mx-1">
+          <img src="${
+            show.image !== null ? show.image.medium : ""
+          }" style="height: 250px" class="card-img-top mx-0" alt="...">
+          <div class="card-body mb-4">
+          <h5 class="text-primary">${show.name}</h5>
+          <span class="badge text-bg-dark">Rated: ${show.rating.average}</span>
+          <span class="badge text-bg-dark">Genres: ${show.genres}</span>
+          <span class="badge text-bg-dark">Status: ${show.status}</span>
+          <span class="badge text-bg-dark">Runtime: ${show.runtime}</span>
+          <p>${show.summary}</p>
+          </div>
+        </div>
+      </div>
+    `;
+  });
+  showEpisodseRow.innerHTML = output;
+
+  // Create option for each show in select after loading page
+  allShows.forEach((show) => {
+    const option = document.createElement("option");
+    option.value = show.name;
+    option.textContent = show.name;
+    selectShows.appendChild(option);
+  });
+}
+
+// Search specific Shows based on select
+async function selectShow(event) {
+  if (event.target.value == "allSerials") {
+    setup();
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    searchUser.value = "";
+    searchUser.disabled = true;
+  } else {
+    searchUser.value = "";
+    searchUser.disabled = false;
+    //Remove options from episodes's select
+    const options = selectEpisodes.getElementsByTagName("option");
+    for (let i = options.length - 1; i > 0; i--) {
+      selectEpisodes.removeChild(options[i]);
+    }
+    // Find selected movie
+    const specificShow = allShows.find(
+      (show) => show.name === event.target.value
+    );
+    // Determine show's id
+    const specificShowID = specificShow.id;
+    // Get all show's episodes
+    allEpisodes = await getAllAvailableEpisodes(specificShowID);
+    // make card for each episode
+    makeCardForEpisodes(allEpisodes);
+  }
 }
 
-window.onload = setup;
+//Get all episodes via fetch
+async function getAllAvailableEpisodes(showId) {
+  // await response of the fetch call
+  const response = await fetch(
+    `https://api.tvmaze.com/shows/${showId}/episodes`
+  );
+  // only proceed once its resolved
+  const data = await response.json();
+  // only proceed once second promise is resolved
+  return data;
+}
+
+//make card for each episode
+function makeCardForEpisodes(episodeList) {
+  let output = "";
+  episodeList.forEach((episode) => {
+    output += `
+      <div class="col">
+        <div class="card h-100 px-0 mx-1">
+        <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
+        <div class="card-body mb-4">
+        <h5 >${episode.name} - S${episode.season
+      .toString()
```

Comment:
I think this multi-line expression inside a `${}` is quite hard to read - I'd think about extracting a variable for this. Something like:

```js
  episodeList.forEach((episode) => {
    let paddedSeasonNumber = episode.season.toString().padStart(2, "0");
    let paddedEpisodeNumber = episode.season.toString().padStart(2, "0");
    let episodeIdentifier = `S${paddedSeasonNumber}E${paddedEpisodeNumber}`;
    output += `
      <div class="col">
        <div class="card h-100 px-0 mx-1">
          <img src="${episode.image.medium}" class="card-img-top mx-0" alt="...">
          <div class="card-body mb-4">
            <h5>${episode.name} - ${episodeIdentifier}</h5>
            <p>${episode.summary}</p>
          </div>
          <a href="${episode.url}" class="btn btn-primary" style="position:absolute; bottom:10px; left:20px; right:20px;" target="_blank">See More</a>
        </div>
      </div>
    `;
  });
```

I think this makes it much easier to understand what's going on - we have some good clear names for each part that's going into the identifier, and we can see in our `h5` that we're just putting the episode identifier into the heading.

Reply:
Thanks a lot.  I've done it. 🙏